frontend/src/components/map/LeafletMap.tsx

```
@@ -1,117 +1,346 @@
+    data &&
+    typeof data.latitude === 'number' &&
+    typeof data.longitude === 'number' &&
+    typeof data.pm2_5_prediction === 'number' &&
+    typeof data.timestamp === 'string'
+  );
+};
+
+// Update the air quality info function to handle invalid values
+const getAirQualityInfo = (pm25: number | null) => {
+  // Handle invalid or null PM2.5 values
+  if (pm25 === null || isNaN(pm25)) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)